Tests/ResolveProjectFolders.Test.ps1

```diff
@@ -0,0 +1,199 @@
+        AddDependency -app 'testApp2' -dependencyAppId $anyAppId -dependencyAppName 'Any'
+
+        AddDependency -app 'bcptApp1' -dependencyAppId $apps['app1'].id -dependencyAppName $apps['app1'].name
+        AddDependency -app 'bcptApp1' -dependencyAppId '75f1590f-55c5-4501-ae63-bada5534e852' -dependencyAppName 'Performance ToolKit'
+
+        AddDependency -app 'bcptApp2' -dependencyAppId $apps['app1'].id -dependencyAppName $apps['app1'].name
+        AddDependency -app 'bcptApp2' -dependencyAppId $apps['app2'].id -dependencyAppName $apps['app2'].name
+        AddDependency -app 'bcptApp2' -dependencyAppId '75f1590f-55c5-4501-ae63-bada5534e852' -dependencyAppName 'Performance ToolKit'
+    }
+
+    It 'When no app folders specified, scans all folders and finds all apps' {
+        $projectSettings = @{ 'appFolders' = @(); 'testFolders' = @(); 'bcptTestFolders' = @()}
+        ResolveProjectFolders -baseFolder $baseFolder -project $ALGoProject -projectSettings ([ref] $projectSettings)
+
+        $projectSettings.appFolders | Should -HaveCount 2
+        $projectSettings.appFolders | Should -Contain 'app1'
+        $projectSettings.appFolders | Should -Contain 'app2'
+
+        $projectSettings.testFolders | Should -HaveCount 2
+        $projectSettings.testFolders | Should -Contain 'testApp1'
+        $projectSettings.testFolders | Should -Contain 'testApp2'
+
+        $projectSettings.bcptTestFolders | Should -HaveCount 2
+        $projectSettings.bcptTestFolders | Should -Contain 'bcptApp1'
+        $projectSettings.bcptTestFolders | Should -Contain 'bcptApp2'
+    }
+
+    It 'WHen only appFolders specified, scan only app folders and resolves them' {
+        $projectSettings = @{ 'appFolders' = @('app1'); 'testFolders' = @(); 'bcptTestFolders' = @()}
+
+        ResolveProjectFolders -baseFolder $baseFolder -project $ALGoProject -projectSettings ([ref] $projectSettings)
+        
+        $projectSettings.appFolders | Should -HaveCount 1
+        $projectSettings.appFolders | Should -Contain 'app1'
+
+        $projectSettings.testFolders | Should -HaveCount 0
+
+        $projectSettings.bcptTestFolders | Should -HaveCount 0
+    }
+
+    It 'When appFolders contains non-existing apps, they are not resolved' {
+        $projectSettings = @{ 'appFolders' = @('nonExistingApp', 'app1'); 'testFolders' = @('nonExistingTestApp'); 'bcptTestFolders' = @('nonExistingBCPTApp')}
+
+        ResolveProjectFolders -baseFolder $baseFolder -project $ALGoProject -projectSettings ([ref] $projectSettings)
+        
+        $projectSettings.appFolders | Should -HaveCount 1
+        $projectSettings.appFolders | Should -Contain 'app1'
+
+        $projectSettings.testFolders | Should -HaveCount 0
+
+        $projectSettings.bcptTestFolders | Should -HaveCount 0
+    }
+
+    It 'When only non-existing apps are specified, scans all folders and finds all apps' {
```

Comment:
Isn't this a change in behavior?
Earlier, if you specified one of either appFolders, testFolders or bcptTestFolders, then you were assumed to be "on your own" and no auto-resolve would kick in.

I guess the new behavior won't break anything as it only kicks in if all folders doesn't resolve to anything and as such wouldn't run anything, but in the case where people are switching from Windows to Linux and folders are case sensitive, then you might go from building 1 app to building more apps without understanding why.

Reply:
The current/existing behavior isn't changed. So, if that causes issues when switching from Windows to Linux, that's another problem.

Reply:
I disagree - it has changed.
Example:
![image](https://github.com/microsoft/AL-Go/assets/10775043/8b51473c-5852-4a72-935b-3aa75edf50fe)

Earlier, if you had "appFolders": [ "nonexisting" } - we would NOT scan for appfolders - we would think that you know what you are doing.

Now we will scan for appFolders.

I think the default behavior of scanning for folders is fine if you haven't specified folders yourself - but if you have specified folders - we should not scan

Reply:
I guess the first attempt to fix the issue with project dependencies was [this commit](https://github.com/microsoft/AL-Go/commit/4134d4aa838aed0733432e40fd541666db99c6b2). Since then, the behavior hasn't changed.

Reply:
In that commit - it only scans for appfolders if settings are not set - and this is how it is in preview today:
![image](https://github.com/microsoft/AL-Go/assets/10775043/60bde4c3-fa87-4940-be37-196f8e358e92)

My point is that the default behavior should only happen if you haven't set the setting yourself.
Default behavior should not kick in if you have misspelled a folder name.

Reply:
Ok, I think we referred to different process: one is when determining project dependencies and one is when determining the AL projects to built.

Both these processes need to resolve the app folders the same way and with my latest changes it is as you mentioned: only scan for app folders of none are defined.

Reply:
Funny thing: the tests failed because they are run on linux in the CI tests. Now they pass both locally (windows) and in the CI (linux).